sky/utils/log_utils.py

```diff
@@ -191,3 +193,31 @@ def readable_time_duration(start: Optional[float],
         diff = diff.replace('hour', 'hr')
 
     return diff
+
+
+def create_and_symlink_log_dir(log_dir: str):
+    """Create a log directory and symlink to it from parent directory.
+
+    If file operations fail, will log a warning and return without error.
+
+    Args:
+        log_dir (str): Expanded log directory path.
+    """
+    os.makedirs(log_dir, exist_ok=True)
+    symlink_path = os.path.join(os.path.dirname(os.path.abspath(log_dir)),
+                                constants.SKY_LATEST_LINK)
+    if os.path.exists(symlink_path):
+        if os.path.islink(symlink_path):
+            try:
+                os.remove(symlink_path)
+            except OSError:
+                return
```

Comment:
Should we add a `logger.warning(...)` here and at L223 to surface the error message?

Reply:
I think it would make sense here. Maybe not for L223 since some upstream callers log to `/dev/null`, which would cause nuisance warnings. Although maybe that should be fixed upstream? Let me know what you think.

Reply:
Agreed, we do not want nuisance warnings. Maybe we can special case for `/dev/null` at L223 and not print the warning in those cases?

Reply:
Yup, just pushed that change!